AK/StdLibExtras.h

```diff
@@ -90,16 +90,28 @@ constexpr SizeType array_size(T (&)[N])
     return N;
 }
 
-template<typename T>
-constexpr T min(T const& a, IdentityType<T> const& b)
+constexpr auto min(auto&& a, auto&& b)
+requires IsSame<RemoveCVReference<decltype(a)>, RemoveCVReference<decltype(b)>>
+{
+    return b < a ? forward<decltype(b)>(b) : forward<decltype(a)>(a);
```

Comment:
I'm not sure why these `forward<decltype(...)>(...)` are needed. It seems to compile fine without.

Reply:
to avoid a copy if the arg is movable